Sources/CNIODarwin/shim.c

```diff
@@ -50,4 +50,25 @@ int CNIODarwin_recvmmsg(int sockfd, CNIODarwin_mmsghdr *msgvec, unsigned int vle
     errx(EX_SOFTWARE, "recvmmsg shim not implemented on Darwin platforms\n");
 }
 
+struct cmsghdr *CNIODarwin_CMSG_FIRSTHDR(const struct msghdr *mhdr) {
+    return CMSG_FIRSTHDR(mhdr);
+}
+
+struct cmsghdr *CNIODarwin_CMSG_NXTHDR(const struct msghdr *mhdr, const struct cmsghdr *cmsg) {
+    return CMSG_NXTHDR(mhdr, cmsg);
+}
+
+const unsigned char *CNIODarwin_CMSG_DATA(const struct cmsghdr *cmsg) {
```

Comment:
Can we add assertions that the pointers passed to these three functions aren't `NULL`?

Reply:
Will do - interestingly it is valid to pass `cmsg = NULL` to `CMSG_NXTHDR` on Darwin but not on Linux.